### DISCUSSION TITLE
Crowbar opens locked cabinets (steel secure, toxin)

### DISCUSSION THREAD

**Discussion by Veritius** [FakeFishGames/Barotrauma#4683] [Ideas and suggestions]
Possibly allow the crowbar to open ID-specific storages such as Secure Steel Cabinet and Toxin Cabinet.
Additionally, the sub designer to enable or disable this functionality and alter the time required.

**Comment by DreamerOfRain** [comment]
Best is adding an equipped required field similar to the button door: 
![image](https://user-images.githubusercontent.com/70696707/104095157-e0e4e980-52c7-11eb-903f-4ae639d53673.png)

**Comment by DeinococcusR** [comment]
Would also be cool if those secure cabinets can be blasted open by explosives to get the precious contents.

**Comment by NilanthAnimosus** [comment]
Explosions can already target items to break them, Perhaps have an item that can repair sealed containers and allow broken containers to be opened by anybody (They don't drop their contents, just lose their access protection).

That way we could use the "Sabotage" feature or certain tools/items to target and damage these locked containers to gain (or grant others) access to them.

In small MP campaigns, having a weapons locker my trusted friends cannot access would be nice to just break the lock proverbially speaking and leave broken.

It should definitely not be fast, and should be fairly noisy to break one of these open, And a noiseless method if available be even slower to do, should it be implemented.

**Reply by Solbusaur** [reply]
I think the sabotage option should be the only real way to open a cabinet. Or perhaps a success/fail chance for a crowbar, giving the player 
 moderate amounts of blunt force trauma or lacerations if they fail

**Reply by mLuby** [reply]
Broken crowbar if they fail.

🔓 What about adding a separate single slot to Containers that can only hold a `Padlock` item? There could even be different types—like an ID-based lock, a 🔐 physical key lock, a 🔏 code lock, etc. Players can only interact with the Padlock if it's locked, but can interact with the whole container if the Padlock is open.

The main advantages of Padlocks is players could target and try to damage/pick the _Padlock_ rather than the container itself. And it's in-game rather than set in stone from the Sub Editor (though that can still be done). And players could add their own Padlock to a small personal cabinet if they've earned that privilege on the sub. Of course, Padlocks would also create new low-key griefing options, which can be both a negative and a positive in a permissive game like Barotrauma.

**Reply by NilanthAnimosus** [reply]
Trying to interact with a stored item without taking it out of the object is a fairly...interestingly awkward thing to propose as far as I know, then couple it with distinguishing the slot from others and not making it more complex - generally these are handled by required tag's on ID's and specific checks in the code, the idea is nice but I don't think it should be an item you place in a slot.

perhaps instead having another item (Like a tool) apply and consume some unspecified item, with another item capable of removing it if you had access re-making said item (Or losing it). but at that point it might as well be just standard ID locks and a way to simply set one up and remove/bypass (With time) it at-will during a round (With some restrictions due to griefing on its usage).

**Reply by mLuby** [reply]
Fair points.

A different approach to the same goal is to have a **wall-attachable Padlock** with a `lock_out` pin, and give containers a `lock_in` pin so players could wire them up to Padlocks. As long as the attached Padlock is sending an "enabled" signal:
1. the Padlock's wiring can't be modified*,
2. the thing the Padlock is wired to can't have its wiring modified*, and
3. the thing the Padlock is wired to can't be opened†.

Ideally there'd be both 🔐 physical locks and 📟 electronic locks.

\* `ConnectionPanel.pickable_in_game = false`
† `ItemContainer.pickable_in_game = false`
